### DISCUSSION TITLE
How do I submit changes to an open source repo?

### DISCUSSION THREAD

**Discussion by tzmanics** [community/community#22430] [New to GitHub]
<p>If I want to add or fix something in a project that is not mine, or contribute to an open source project what are the steps I take?</p>
<p>Thanks!</p>

**Comment by a-a-ron** [comment]
<p>This is a great question. GitHub has a <a href="https://guides.github.com/activities/forking/" rel="nofollow noopener">guide</a> with step-by-step instructions to help someone through the process.  </p>

**Comment by Ben3eeE** [comment, marked as answer]
<p>There is an amazing <a href="https://opensource.guide/" title="Open source guides" rel="nofollow noopener">Open Source Guide</a> created by GitHub and friends that contains all the information you need to know about contributing to an open source project and more! The specific section you are looking for in this guide is <a href="https://opensource.guide/how-to-contribute/" title="How to Contribute to Open Source" rel="nofollow noopener">How to Contribute to Open Source</a>.</p>
<p>What is great about this guide is that it is <a href="https://github.com/github/opensource.guide" title="GitHub opensource.guide repository" rel="nofollow noopener">an open source project</a> so if you find an error or want to improve on it you can by creating an issue or opening a pull request.</p>

**Comment by Rous110508** [comment]
<aside class="quote" data-post="1" data-topic="280">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/how-do-i-submit-changes-to-an-open-source-repo/280">How do I submit changes to an open source repo?</a> <a class="badge-wrapper  bullet" href="/c/github-help/how-to-use-git-and-github/21"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #ffd1ac;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Ask and answer questions regarding Git and GitHub.">How to use Git and GitHub</span></a>
  </div>
  <blockquote>
    If I want to add or fix something in a project that is not mine, or contribute to an open source project what are the steps I take? 
Thanks!
  </blockquote>
</aside>

**Comment by shubhransh-gupta** [comment]
Great question — here is the standard open-source contribution flow on GitHub:

## 1) Find an issue first
Look for issues labeled `good first issue`, `help wanted`, or `documentation`.
Comment: **"I would like to work on this — please assign me."**

## 2) Fork the repository
Click **Fork** on the top-right of the repo page.

## 3) Clone your fork locally
```bash
git clone https://github.com/YOUR_USERNAME/REPO.git
cd REPO
git remote add upstream https://github.com/ORIGINAL_OWNER/REPO.git
```

## 4) Create a feature branch
```bash
git checkout -b fix/short-description
```

## 5) Make a focused change
- Keep the PR small (one bug fix or one feature)
- Follow the project's CONTRIBUTING.md
- Run tests/lint if the repo has them

## 6) Commit with a clear message
```bash
git add .
git commit -m "Fix typo in README installation steps"
```

## 7) Push to your fork
```bash
git push -u origin fix/short-description
```

## 8) Open a Pull Request
GitHub will show **Compare & pull request** after push.
In the PR description, add: `Fixes #123` if it closes an issue.

## 9) Respond to review feedback
Maintainers may request changes — push new commits to the same branch.

## 10) After merge
Sync your fork:
```bash
git checkout main
git pull upstream main
git push origin main
```

### First-time contributor tips
- Docs/README fixes are the fastest way to get a merged PR
- Read existing PRs to learn the maintainer's style
- Be patient — reviews can take days

If you share a repo link, I can suggest a good first issue to pick.
